### ISSUE TITLE
Publish the npm provenance data

### ISSUE THREAD

**Issue by nazarhussain** [ChainSafe/lodestar#5423]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

Github recently allowed a new feature to publish the npm package provenance data linked to npm publish page. See the link in the additional context

**Describe alternatives you've considered**

There is no automated alternative for now. 

**Additional context**
https://github.blog/2023-04-19-introducing-npm-package-provenance/

**Comment by philknows** [comment]
Is this a sufficient alternative to closing #3596 ? Also related to our discussion about managing dependencies #3470 .

**Comment by wemeetagain** [comment]
> Is this a sufficient alternative to closing https://github.com/ChainSafe/lodestar/issues/3596

No, #3596 is about mitigating supply-chain attacks on our dependencies.
This is just helpful for lodestar library users, so they can better trust the validity of our libraries (eg `@lodestar/config`)

**Comment by nflaig** [comment]
this might be resolved since we switched to trusted publishing

see https://www.npmjs.com/package/@lodestar/types

<img width="916" height="257" alt="Image" src="https://github.com/user-attachments/assets/3173af2c-b259-4ebb-9edb-ec826607f15a" />

is there anything else we need to do @nazarhussain ?

**Comment by wemeetagain** [comment]
yup, resolved with trusted publishing
